lambdas/process-journey-event/src/test/java/uk/gov/di/ipv/core/processjourneyevent/JourneyMapPageContextTest.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.di.ipv.core.processjourneyevent;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import uk.gov.di.ipv.core.library.domain.IpvJourneyTypes;
+import uk.gov.di.ipv.core.processjourneyevent.statemachine.StateMachineInitializer;
+import uk.gov.di.ipv.core.processjourneyevent.statemachine.states.BasicState;
+import uk.gov.di.ipv.core.processjourneyevent.statemachine.states.State;
+import uk.gov.di.ipv.core.processjourneyevent.statemachine.stepresponses.PageStepResponse;
+import uk.org.webcompere.systemstubs.environment.EnvironmentVariables;
+import uk.org.webcompere.systemstubs.jupiter.SystemStub;
+import uk.org.webcompere.systemstubs.jupiter.SystemStubsExtension;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(SystemStubsExtension.class)
+public class JourneyMapPageContextTest {
+    @SystemStub private static EnvironmentVariables environmentVariables;
+
+    @BeforeAll
+    public static void beforeAll() {
+        environmentVariables.set("IS_LOCAL", "true");
+    }
+
+    private final HashMap<String, List<String>> expectedStateContexts =
+            new HashMap<>() {
+                {
+                    put("no-photo-id", List.of("MITIGATION_02_OPTIONS_WITH_F2F_J7"));
+                }
+            };
+
+    @ParameterizedTest
+    @EnumSource
+    void shouldHandleSameContextForSamePage(IpvJourneyTypes journeyType) throws IOException {
```

Comment:
I think this needs a big comment.
As far as I can see this is testing that anywhere that `MITIGATION_02_OPTIONS_WITH_F2F_J7` is used in the map it should have a context of `no-photo-id` but I'm still not 100% sure even after staring at this code for a while.
Relatedly I'm unsure what this test is supposed to catch exactly. AFAIK we can't add a second `MITIGATION_02_OPTIONS_WITH_F2F_J7` to the journey map so this won't catch people adding that and forgetting the context. It will only catch new contexts if the person adding the new context updates this test but then forgets to include the context in the journey map.

Reply:
Ah yep good point, the test should be changed to not check Events against Contexts, but about PageId against Contexts.

My only thought around adding this test was to avoid somebody removing or changing a context value in the JourneyMapPage, as that would result in the front-end displaying the wrong content, or no content at all.

Perhaps this would be better covered as an integration test anyway. @blakeyp, thoughts?

Reply:
Ok that makes sense, it probably is worth at least having this unit test in there to provide a slight safety net